### PR TITLE
read cpu info from cpuacct.usage_percpu if no cpuacct.usage_all

### PR DESCRIPTION
**Pull request by lijiaocn**
Read cpu info from cpuacct.usage_percpu if no cpuacct.usage_all  #273  

Signed-off-by: lijiaocn <lijiaocn@foxmail.com>

### CODE REVIEW

File: bindings.c [hunk 1]
```diff
@@ -4046,6 +4046,49 @@ static uint64_t get_reaper_age(pid_t pid)
...
+	}
+
+	// usage_str format, 4 cores:
+	// 120190269376599 151483821809073 145465919648605 14450503259981
+	token = strtok(usage_str, delim);
+	while (NULL != token && j < cpucount)  {
+		sscanf(token,"%lu", &usage);
+		cpu_usage[j].user = usage / 1000.0 / 1000 / 1000 * ticks_per_sec;
+		cpu_usage[j].system = 0;
+		token = strtok(NULL, delim);
```
Comment (brauner) [inline bindings.c]: We can't really use `strtok` since it's not thread-safe and `libfuse` runs threaded.

Comment (brauner) [reply bindings.c]: You can copy:
```c
#define lxc_iterate_parts(__iterator, __splitme, __separators)                  \
	for (char *__p = NULL, *__it = strtok_r(__splitme, __separators, &__p); \
	     (__iterator = __it);                                               \
	     __iterator = __it = strtok_r(NULL, __separators, &__p))
```
from https://github.com/lxc/lxc/blob/master/src/lxc/macro.h

Comment (lijiaocn) [reply bindings.c]: Thanks, I have fixed this problem with codes in lxc/macro.h,  at the same time add some codes to check `sscanf`'s return value.

**Comment by brauner** [comment]
A version of this has been merged in another pull request, thank you!

**Pull request closed without merging**